boards: silabs: Add bgm220_ek4314a board

asmellby opened the pull request:
Add BGM220P SoC support and add BGM220 Explorer Kit. The board has a BGM220P module, button and
LED as well as mikroBUS and Qwiic expansion connectors.

This PR is stacked on top of #95731.

asmellby committed:
soc: silabs: Add BGM220P modules

Add support for BGM220P modules. Enable oscillators in SoC DTS
since the necessary crystals are present in the modules.

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

asmellby committed:
boards: silabs: Add bgm220_ek4314a board

Add BGM220 Explorer Kit. The board has a BGM220P module, button and
LED as well as mikroBUS and Qwiic expansion connectors.

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

asmellby committed:
soc: silabs: xg22: Add missing EUART0 peripheral

Add missing EUART0 peripheral to devicetree for xg22.
Fix NUM_IRQS, there are 64 external interrupts on xg22.
Remove `select` of UART_INTERRUPT_DRIVEN at SoC level, this doesn't
belong here, since it prevents disabling the UART. This should be a
board or application level decision.

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

nordicjm reviewed (dismissed):
SoC/board files OK

ydamigos commented on boards/silabs/dev_kits/bgm220_ek4314a/Kconfig.defconfig:
```diff
@@ -0,0 +1,28 @@
+	default 875000
+	depends on LOG_BACKEND_SWO
+
+if SOC_GECKO_USE_RAIL
+
+config FPU
+	default y
+
+endif # SOC_GECKO_USE_RAIL
+
+if BT
+
+config FPU
+	default y
+
+config MAIN_STACK_SIZE
+	default 3072 if PM
+	default 2304
+
+endif # BT
```
```suggestion
config FPU
	default y if SOC_GECKO_USE_RAIL || BT

if BT

config MAIN_STACK_SIZE
	default 3072 if PM
	default 2304

endif # BT
```

asmellby committed:
tests: subsys: logging: Rename emulated uart in log_backend_uart

The emulated uart was named `euart0`, which collides with the name
of a physical uart on some Silicon Labs devices. Since the test
is not limited to run in simulation, rename the emulated uart
node to something less likely to collide with physical uart names.

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

jhedberg commented:
Removed the "DNM (manifest)" label after verifying that the module changes are compliant with Zephyr policy.
